tasks/migrate_agents_nix.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# Ensure we can reach the target Puppet Enterprise server
+echo "Verifying connectivity to target Puppet server $PT_target_pe:8140..."
+if timeout 1 bash -c "cat < /dev/null > /dev/tcp/$PT_target_pe/8140"; then
+    echo "Target Puppet server is reacheable"
+else
+    echo "Target Puppet server is not reacheable, aborting migration!"
+    exit 1
+fi
+
+# Ensure Curl is installed
+if ! command -v curl &> /dev/null; then
+    echo "Curl is not installed on this system, unable to continue!"
+    exit 2
+fi
+
+echo 
+echo "Stopping the Puppet Agent service..."
+service puppet stop 2>&1
+
+if [ $PT_regenerate == 'true' ]; then
+    echo 
+    echo "Regenerate flag detected, clearing out existing node certificates before migration..."
+    localcacert=$(puppet config print localcacert)
+    hostcert=$(puppet config print hostcert)
+    hostprivkey=$(puppet config print hostprivkey)
+    hostpubkey=$(puppet config print hostpubkey)
+    hostcrl=$(puppet config print hostcrl)
+    hostcsr=$(puppet config print hostcsr)
+    declare -a arr=("$localcacert" "$hostcert" "$hostprivkey" "$hostpubkey" "$hostcrl" "$hostcsr")
+    for i in "${arr[@]}"
+    do
+        if [ -f "$i" ]; then
+            echo "Deleting $i..."
+            rm -f $i 2>&1
+        fi
+    done
+    echo "Existing node certificates have been cleared, new certificates will be generated on the next Puppet run"
+fi
+
+echo 
+echo "Pointing Puppet Agent to new PE server..."
+puppet config set server $PT_target_pe
```

Comment:
I think this will be ignored if server_list was previously set and PE configures it in agent section

Reply:
Should we simply add a `puppet config delete server_list` to fix that?

Reply:
probably removing server_list form main and agent sections and then setting server in agent section will work universally

I have a plan that we use for the migration and I just do this
```
  run_task('puppet_conf', $agent_targets,
    action  => 'set',
    section => 'agent',
    setting => 'server_list',
    value   => "${new_server}:8140",
  )
```

Plan is more complicated because it handles signing certificates on the new server and purging them from old ones, so we can track what was migrated and what wasn't
I can send it to you if you like

Reply:
OK, I'll add the commands for clearing those settings.
A plan that interacts with both old and new PE systems is more powerful, but also more difficult to make work as a universal solution. Let's provide these simple tasks first, a more complex plan can always follow later.

Reply:
Sounds good. One more important change I had to do while migrating from PE2018 to PE2019
```
  # Puppet 5 can't validate certificate chain
  run_task('puppet_conf', $agent_targets,
    action  => 'set',
    section => 'main',
    setting => 'certificate_revocation',
    value   => 'leaf',
  )
```

Reply:
Ah that's a good solution to enable migrations from 5.x.
I can add that setting when `puppet --version | grep -e ^5` resolves true.